app/models/test.rb

```diff
@@ -6,9 +6,17 @@ class Test < ApplicationRecord
   has_many :tests_users, dependent: :destroy
   has_many :users
 
-  def self.all_by_category(title)
-    Test.joins(:category)
-        .where('categories.title = ?', title)
-        .pluck(:title)
+  scope :easy, -> { where(level: 0..1) }
+  scope :middle, -> { where(level: 2..4) }
+  scope :hard, -> { where(level: 5..Float::INFINITY) }
+  scope :all_by_category, ->(title) { joins(:category).where('categories.title = ?', title) }
```

Comment:
```suggestion
  scope :all_by_category, ->(title) { joins(:category).where(categories: { title: title }) }
```
в прошлый раз уже предлагал - это рекомендуемый вариант записи